osler/core/forms.py

```diff
@@ -63,7 +71,21 @@ def __init__(self, *args, **kwargs):
         self.helper['languages'].wrap(InlineCheckboxes)
         self.helper['ethnicities'].wrap(InlineCheckboxes)
         self.helper.add_input(Submit('submit', 'Submit'))
-        self.fields['address'].widget.attrs = {'placeholder': settings.OSLER_DEFAULT_ADDRESS}
+        self.fields['address'].widget.attrs = {'placeholder': settings.OSLER_DEFAULT_ADDRESS,
+                                                'id': 'address'}
+        self.fields['city'].widget.attrs = {'id': 'city'}
+        self.fields['state'].widget.attrs = {'id': 'state'}
+        self.fields['zip_code'].widget.attrs = {'id': 'zip code'}
+        self.fields['country'].widget.attrs = {'id': 'country'}
+        self.fields['pcp_preferred_zip'].widget.attrs = {'id': 'pcp preferred zip'}
+        if settings.OSLER_HOMELESS_OPTION:
+            self.helper.layout.remove('homeless')
+            self.helper.layout.insert(14,'homeless')
```

Comment:
I'm not thrilled about this hard-coded index... maybe we should ask for the index of the homeless field first? If ever we reorder the form we'll have a problem tracking down this spot in the code.

As an aside, I wonder if we should be writing a special kind of widget that does this? I know django offers the ability to subclass widget.

Reply:
I've looked pretty hard into the documentation for crispy form, and it seems that we're going to end up having to do some amount of hard-coding... What would your ideal widget do @justinrporter ?

Reply:
So manually adding a new field to crispy forms always appends it at the end. Since the crispy form is generated from the model, we could have the homeless setting alter the model before we load the form so that it's in the right place. But I don't really know if this is bad practice or if its even possible. What do you guys think?